go/ql/lib/semmle/go/Decls.qll

```diff
@@ -649,17 +649,23 @@ class ReceiverDecl extends FieldBase, Documentable, ExprParent {
  * Examples:
  *
  * ```go
+ * int
+ * string
  * error
  * r io.Reader
+ * output string
+ * err error
  * x, y int
```

Comment:
Minor: I note that `string` and `error` are listed separately here, despite appearing in the same example below. Meanwhile `x, y int` are listed together here (obviously) since they appear in the same example below. I found that slightly confusing, but I am guessing it makes sense in terms of how the results are returned?

If we really wanted to improve this documentation more, I would maybe suggest something like:

------

Consider the following function definitions:

```go
func f1() int { return 0 }
func f2(input string) (string, error) { return "", nil }
func f3(a int) (r io.Reader) { return nil }
func f4(input string) (output string, err error) { return}
func f5(e error) (x, y int) { return }
```

The query:

```ql
from FuncTypeExpr f, ResultVariableDecl rvd, 
where rvd = f.getAResultDecl()
select f, rvd
```

Would return:

```
func f1() int { return 0 }, int
.... 
```

Reply:
Could we at least add a note to the comment then to make this more explicit? I don't think it would be very nice to our users (and indeed ourselves!) if the intended way to understand the example requires writing an example query to test it!